engine/src/main/java/org/terasology/logic/inventory/ItemPickupAuthoritySystem.java

```diff
@@ -59,20 +66,32 @@ public void onDropItemEvent(DropItemEvent event, EntityRef itemEntity, ItemCompo
         if (!itemEntity.hasComponent(LocationComponent.class)) {
             itemEntity.addComponent(new LocationComponent(event.getPosition()));
         }
+
+        itemEntity.getComponent(ItemComponent.class).pickableUp = false;
+        long timeToPickUp = itemEntity.getComponent(ItemComponent.class).pickUpTime;
+        delayManager.addDelayedAction(itemEntity, DELAYED_PICKUP_ACTION_ID, timeToPickUp);
     }
 
+    @ReceiveEvent
+    public void onDelayedPickup(DelayedActionTriggeredEvent event, EntityRef entityRef) {
+        if (event.getActionId().equals(DELAYED_PICKUP_ACTION_ID)) {
+            entityRef.getComponent(ItemComponent.class).pickableUp = true;
+        }
+    }
 
     @ReceiveEvent
     public void onBumpGiveItemToEntity(CollideEvent event, EntityRef entity, PickupComponent pickupComponent) {
-        GiveItemEvent giveItemEvent = new GiveItemEvent(event.getOtherEntity());
-        entity.send(giveItemEvent);
+        if ( entity.getComponent(ItemComponent.class).pickableUp == true ) {
```

Comment:
Couldn't I just do as above and only use onBumpGiveItemToEntity when the ItemComponent is present?